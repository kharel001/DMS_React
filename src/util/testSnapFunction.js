import React from 'react';
import {
    configure,
    shallow
} from 'enzyme';
import EnzymeAdapter from 'enzyme-adapter-react-16';
// JestHook.mock('expo-font');
configure({
    adapter: new EnzymeAdapter
})


export  const findByTestAttr=(wrapper, val)=>{
    return wrapper.find(`[data-test='${val}']`)
}

export const findByTestProps=(wrapper,val)=>{
    return wrapper.findByProps({"data-test": val})
}

//  jest.mock("Platform",()=>"android")
export const mockPlatform = OS => {
    jest.resetModules();
    jest.doMock('react-native/Libraries/Utilities/Platform', () => ({
      OS,
      select: config => config[OS],
    }));
  };
export const mockAlert = ()=>{
    jest.resetModules();
    jest.doMock('react-native/Libraries/Alert/Alert', () => ({
        alert:jest.fn()
      }));
    };

export const mockDevice = ()=>{
    jest.resetModules();
    jest.doMock('react-device-detect', () => ({
        isIOS:false,
        isMobile:true
      }));
    };
const useHistoryFunc=jest.fn();
const listenFunc = jest.fn();
useHistoryFunc.mockReturnValue({
    listen:listenFunc
})
export const mockHistory = ()=>{
    jest.resetModules();
    // jest.doMock('react-router-dom/cjs/react-router-dom', () => ({
    //     useHistory: listenFunc
    //   }));

    };
export default function testSnapFunction(oDesc, descriptions,element){
    describe(oDesc, () => {

        afterEach(() => {
            jest.clearAllMocks();
          });

        test(descriptions, () => {
            const tree = shallow(element);
            expect(tree).toMatchSnapshot();
        });
    
    })
}

