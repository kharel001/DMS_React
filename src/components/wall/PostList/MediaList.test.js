
import React from 'react';
import MediaList from './MediaList';
import testSnapFunction from '../../../util/testSnapFunction';
testSnapFunction("<MediaList/>", "Snapshot test for MediaList",<MediaList /> )
