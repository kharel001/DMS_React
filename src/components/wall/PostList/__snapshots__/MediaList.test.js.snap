// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MediaList/> Snapshot test for MediaList 1`] = `ShallowWrapper {}`;
